cmd/ipfs/dist/install.sh

```diff
@@ -2,20 +2,36 @@
 
 bin=ipfs
 
-# this script is currently brain dead.
-# it merely tries two locations.
-# in the future maybe use value of $PATH.
+old_IFS="$IFS"
+IFS=':'
 
-binpath=/usr/local/bin
-if [ -d "$binpath" ]; then
-  mv "$bin" "$binpath/$bin"
-  echo "installed $binpath/$bin"
-  exit 0
-fi
+echo "Welcome to the $bin installer."
+echo "With your permission it will try to install the $bin binary into one of"
+echo "the folders found in your \$PATH variable."
+echo
 
-binpath=/usr/bin
-if [ -d "$binpath" ]; then
-  mv "$bin" "$binpath/$bin"
-  echo "installed $binpath/$bin"
-  exit 0
+if [ ! -f "$bin" ]; then
+  echo "The '$bin' binary was not found in the current directory."
+  echo "There is nothing to install."
+  exit 1
 fi
+
+# Make sure that /usr/local/bin and /usr/bin are the first paths we try.
+# Note that these paths might be already found in $PATH.
+paths="/usr/local/bin:/usr/bin:$PATH"
+
+for binpath in $paths;
+do
+  if [ -d "$binpath" ]; then
```

Comment:
@jbenet At this point we might want to ask the user whether they would really like to install `ifps` to given directory `$binpath`. I am sure there are people who would like to skip `/usr/bin/local` and rather install `ipfs` to `~/bin` for instance.